tools/OloHeaderTool/main.cpp

```
@@ -0,0 +1,842 @@
+                //   "glm::vec3 m_Color = { 1.0f, 1.0f, 1.0f };"
+                //   "f32 m_Intensity = 1.0f;"
+                //   "SceneCamera Camera;"
+                std::string fieldType;
+                std::string fieldName;
+                {
+                    std::istringstream tokens(trimmed);
+                    tokens >> fieldType;
+                    tokens >> fieldName;
+                    auto cutPos = fieldName.find_first_of("={;");
+                    if (cutPos != std::string::npos)
+                        fieldName = fieldName.substr(0, cutPos);
```

Cppcheck (2.20.0)

[performance] Ineffective call of function 'substr' because a prefix of the string is assigned to itself. Use resize() or pop_back() instead.

(uselessCallsSubstr)